### PR TITLE
[GPU] Pass Resample unit tests on DG2

### PR DESCRIPTION
**Pull request by tuxedcat**
### Details:
 - Change Scale to Eltwise( mode::prod ) in unit test topologies.
 - Support Resample-Eltwise fusing.

### Tickets:
 - 67491

### CODE REVIEW

File: src/plugins/intel_gpu/tests/fusions/resample_fusion_test.cpp [hunk 1]
```diff
@@ -252,86 +252,88 @@ INSTANTIATE_TEST_SUITE_P(fusings_gpu, resample_quantize_concat, ::testing::Value
...
+class resample_eltwise_concat : public ResamplePrimitiveFusingTest {};
+TEST_P(resample_eltwise_concat, along_f) {
     auto p = GetParam();
     create_topologies(
         input_layout("input", get_input_layout(p)),
         resample("resample1", "input", p.out_shape, p.in_shape.feature[0], p.type),
-        data("scale1_scale", get_mem(get_per_channel_layout(p), -10, 10)),
-        data("scale1_shift", get_mem(get_per_channel_layout(p), -10, 10)),
-        scale("scale1", "resample1", "scale1_scale", "scale1_shift"),
+        data("eltwise1_eltwise", get_mem(get_per_channel_layout(p), -10, 10)),
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/resample_fusion_test.cpp:261]: eltwise1_eltwise -> eltwise1_scale

File: src/plugins/intel_gpu/tests/fusions/resample_fusion_test.cpp [hunk 2]
```diff
@@ -252,86 +252,88 @@ INSTANTIATE_TEST_SUITE_P(fusings_gpu, resample_quantize_concat, ::testing::Value
...
+        data("eltwise2_eltwise", get_mem(get_per_channel_layout(p), -10, 10)),
+        data("eltwise2_shift", get_mem(get_per_channel_layout(p), -10, 10)),
+        eltwise("eltwise2_prod", { "resample2", "eltwise2_eltwise" }, eltwise_mode::prod),
+        eltwise("eltwise2", { "eltwise2_prod", "eltwise2_shift" }, eltwise_mode::sum),
+        concatenation("concat", { "eltwise1", "eltwise2" }, 1),
         reorder("reorder_bfyx", "concat", cldnn::format::bfyx, p.default_type)
     );
 
-    tolerance = 1e-5f;
+    tolerance = 1e-2f;
```
[the remark this answers is not in the document]
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/resample_fusion_test.cpp]: It's my mistake. I thought scale works like x*scale+bias, but actually it works like (x+bias)*scale. Changed the biased scale testcases.

File: src/plugins/intel_gpu/tests/fusions/resample_fusion_test.cpp (hunk 1 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/resample_fusion_test.cpp]: Done.

File: src/plugins/intel_gpu/tests/fusions/resample_fusion_test.cpp [hunk 3]
```diff
@@ -252,86 +252,88 @@ INSTANTIATE_TEST_SUITE_P(fusings_gpu, resample_quantize_concat, ::testing::Value
...
+        data("eltwise1_eltwise", get_mem(get_per_channel_layout(p), -10, 10)),
+        data("eltwise1_shift", get_mem(get_per_channel_layout(p), -10, 10)),
+        eltwise("eltwise1_prod", { "resample1", "eltwise1_eltwise" }, eltwise_mode::prod),
+        eltwise("eltwise1", { "eltwise1_prod", "eltwise1_shift" }, eltwise_mode::sum),
         resample("resample2", "input", p.out_shape, p.in_shape.feature[0], p.type),
-        data("scale2_scale", get_mem(get_per_channel_layout(p), -10, 10)),
-        data("scale2_shift", get_mem(get_per_channel_layout(p), -10, 10)),
-        scale("scale2", "resample2", "scale2_scale", "scale2_shift"),
-        concatenation("concat", { "scale1", "scale2" }, 1),
+        data("eltwise2_eltwise", get_mem(get_per_channel_layout(p), -10, 10)),
```
[the remark this answers is not in the document]
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/resample_fusion_test.cpp]: Done.

File: src/plugins/intel_gpu/tests/fusions/resample_fusion_test.cpp [hunk 4]
```diff
@@ -252,86 +252,88 @@ INSTANTIATE_TEST_SUITE_P(fusings_gpu, resample_quantize_concat, ::testing::Value
...
+
+    resample_test_params{ CASE_RESAMPLE_I8_1, 3, 8 },
+    resample_test_params{ CASE_RESAMPLE_I8_2, 3, 8 },
+    resample_test_params{ CASE_RESAMPLE_I8_3, 3, 8 },
+    resample_test_params{ CASE_RESAMPLE_I8_4, 3, 8 },
+
+    resample_test_params{ CASE_RESAMPLE_U8_1, 3, 8 },
+    resample_test_params{ CASE_RESAMPLE_U8_2, 3, 8 },
+    resample_test_params{ CASE_RESAMPLE_U8_3, 3, 8 },
+    resample_test_params{ CASE_RESAMPLE_U8_4, 3, 8 },
```
[the remark this answers is not in the document]
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/resample_fusion_test.cpp]: I handled it like this:
```
#define [TEST_CLASS_NAME]_CNT 3, 8
```

**Review by isanghao — APPROVED** [review]

**Pull request merged**